dags/utils/inventory.py

```diff
@@ -0,0 +1,112 @@
+import boto3
+import json
+import gzip
+import csv
+import os.path as op
+from copy import deepcopy
+from datetime import datetime, timedelta
+from airflow.contrib.hooks.aws_hook import AwsHook
+
+from urllib.parse import urlparse
+
+
+class s3:
+    def __init__(self, url, conn, region, suffix=''):
+        self.url = url
+        self.suffix = suffix
+        aws_hook = AwsHook(aws_conn_id=conn)
+        cred = aws_hook.get_session().get_credentials()
+        self.s3 = boto3.client('s3', aws_access_key_id=cred.access_key ,
+                                aws_secret_access_key=cred.secret_key  ,
+                                region_name=region)
+        self.bucket = self.urlparse(self.url)['bucket']
+
+    # Modified derived from https://alexwlchan.net/2018/01/listing-s3-keys-redux/
+    def find(self, url, sub_key):
+        """
+        Generate objects in an S3 bucket.
+        :param: sub_key: string to be present in the object name
+        :param url: The URL of the bucket
+        """
+
+        parts = self.urlparse(url)
+        kwargs = {'Bucket': parts['bucket']}
+
+        while True:
+            # The S3 API response is a large blob of metadata.
+            # 'Contents' contains information about the listed objects.
+            resp = self.s3.list_objects_v2(**kwargs)
+            try:
+                contents = resp['Contents']
+            except KeyError:
+                return
+
+            for obj in contents:
+                key = obj['Key']
+                if sub_key in key and key.endswith(self.suffix):
+                    yield f"s3://{parts['bucket']}/{obj['Key']}"
+
+            # The S3 API is paginated, returning up to 1000 keys at a time.
+            # Pass the continuation token into the next response, until we
+            # reach the final page (when this field is missing).
+            try:
+                kwargs['ContinuationToken'] = resp['NextContinuationToken']
+            except KeyError:
+                break
+
+    def latest_manifest(self):
+        """
+        Return a dictionary of a manifest file"
+        """
+        parts = self.urlparse(self.url)
+        # get latest manifest file
+        today = datetime.now()
+        manifest_url = None
+        for dt in [today, today - timedelta(1)]:
+            _key = op.join(parts['key'], dt.strftime('%Y-%m-%d'))
+            _url = 's3://%s' % (parts['bucket'])
```

Comment:
How about using `f"s3://{parts['bucket']}"` as used elsewhere?

Reply:
Both are correct but changed for consistency.